presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/InstallationManager.java

```diff
@@ -201,88 +289,45 @@ public Response statusUsingRpm()
         }
     }
 
-    private static boolean isInstalled()
-            throws PrestoManagerException
+    private void PrestoAsyncCommand(PrestoRunnable runnable)
     {
-        return executeCommand("rpm", "-q", "presto-server-rpm") == 0;
+        new Thread(() -> {
+            try {
+                runnable.run();
+            }
+            catch (PrestoManagerException e) {
+                LOGGER.error(e.getCause(), e.getMessage());
+            }
+        }).start();
     }
 
-    private static boolean isRunning()
-            throws PrestoManagerException
-    {
-        return isInstalled() && executeCommand("service", "presto", "status") == 0;
-    }
+    abstract void installAsync(URL url, boolean checkDependencies)
+            throws PrestoManagerException;
 
-    public static void terminatePresto()
-            throws PrestoManagerException
-    {
-        int prestoTerminate = executeCommand("service", "presto", "stop");
-        if (prestoTerminate != 0) {
-            throw new PrestoManagerException("Failed to stop presto", prestoTerminate);
-        }
-    }
+    abstract void uninstallAsync(boolean checkDependencies)
+            throws PrestoManagerException;
 
-    private void killPresto()
-            throws PrestoManagerException
-    {
-        executeCommand("sudo", LAUNCHER_SCRIPT, "kill");
-        int prestoKill = executeCommand("service", "presto", "status");
-        if (prestoKill != 3) {
-            throw new PrestoManagerException("Failed to stop presto", prestoKill);
-        }
-    }
+    abstract void upgradeAsync(URL url, boolean checkDependencies, boolean preserveConfig)
+            throws PrestoManagerException;
 
-    private void gracefulStop()
-            throws PrestoManagerException
-    {
-        try {
-            String prestoPort = getPrestoPort();
-            UriBuilder uriBuilder = fromUri(format("http://localhost:%s", prestoPort)).path("/v1/info/state");
-            Response response = client.target(uriBuilder.build()).request(TEXT_PLAIN)
-                    .buildPut(entity(GSON.toJson("SHUTTING_DOWN"), APPLICATION_JSON)).invoke();
-            if (response.getStatus() != 200) {
-                throw new PrestoManagerException("Failed to stop presto gracefully");
-            }
-        }
-        catch (ProcessingException e) {
-            LOGGER.warn(e, "Presto is not running");
-        }
-        catch (IOException e) {
-            throw new PrestoManagerException("Failed to get Presto port number.", e);
-        }
-    }
+    abstract void startAsync()
+            throws PrestoManagerException;
 
-    private static String getPrestoPort()
-            throws IOException
-    {
-        return getFileProperty(PRESTO_CONFIG_FILE, "http-server.http.port");
-    }
+    abstract void terminatePresto()
```

Comment:
`terminatePresto`, `killPresto`, and `gracefulStop` should have more similar names.

I might just call them `terminate`, `kill`, and `stopGracefully` (or maybe`stop`).